extensions-core/druid-basic-security/src/main/java/org/apache/druid/security/basic/authentication/db/updater/NoopBasicAuthenticatorMetadataStorageUpdater.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.security.basic.authentication.db.updater;
+
+import org.apache.druid.security.basic.authentication.entity.BasicAuthenticatorCredentialUpdate;
+import org.apache.druid.security.basic.authentication.entity.BasicAuthenticatorUser;
+
+import javax.annotation.Nullable;
+import java.util.Collections;
+import java.util.Map;
+
+public class NoopBasicAuthenticatorMetadataStorageUpdater implements BasicAuthenticatorMetadataStorageUpdater
```

Comment:
Please add a short doc about
1) whether anybody is supposed to use this class (including via node configuration) or it's solely for use in `BasicSecurityDruidModule`
2) What does specifically "noop" mean here - reject, allow, etc.

Reply:
1. these classes should be used if it is supposed that a node does not support metadata update
2. "noop" means no operation, no exception, empty values if something should be returned

Reply:
Sorry, but this doesn't make sense to me. These comments just repeat the code below. I don't actually understand what does "node supporting/not supporting authenticator metadata storage update" mean. Too many noun words in a row, and without connection to the real world.

I think it would be more understandable if was explained by example, e. g. "Druid's Xxx (broker/coordinator/router/historical/whatever) nodes use this because when requests come from them from nodes Yyy they are already authorized, so these nodes don't need to send delete/create users"... - something like this, a small story. Could you please rewrite the text like this?

The problem stems from `BasicAuthenticatorMetadataStorageUpdater` itself which class-level Javadoc comment just repeats the code without giving understanding. Could you please upgrade its Javadoc? (Then Javadocs for Noop classes can be shorter and refer to `BasicAuthenticatorMetadataStorageUpdater` instead.)

Reply:
`NoopBasicAuthenticatorMetadataStorageUpdater` was developed only to remove nulls from `BasicSecurityDruidModule` with the assumption that its methods won't be called. I don't think the class would be useful except for this case.
It is better to remove the comment `The class is needed if a node should not support authenticator metadata storage update` because it is not correct in general. But again I don't know how to use this class except how it is used now.

Reply:
Consider this optional, but I still don't understand what do `*AuthenticatorMetadataStorageUpdater` classes do, reading through their Javadocs. To me, their Javadocs are currently not reaching their goal of explaining things.